pkg/kritis/gcbsigner/gcb_event_parser.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package gcbsigner
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/grafeas/kritis/pkg/kritis/constants"
+
+	"cloud.google.com/go/pubsub"
+)
+
+// ExtractBuildProvenanceFromEvent extracts the build provenance from a Cloud
+// Builder event.
+// Return the list of images built and their build provenance.  If the event
+// does contain relevant buikld info (e.g., the build is not yet complete, or
+// no images were produced) then 'nil' will be returned.
+//
+// TODO this should validate the provenance in the pubsub message against the
+// information in Container Analysis that is created by Cloud Builder.
+func ExtractBuildProvenanceFromEvent(msg *pubsub.Message) ([]BuildProvenance, error) {
+	var event BuildEvent
+	if err := json.Unmarshal(msg.Data, &event); err != nil {
+		return nil, err
+	}
+	glog.Infof("build %q, status: %q", event.ID, event.Status)
+	glog.Infof("messages: %q", msg.Data)
+	if event.Status == "SUCCESS" {
```

Comment:
Keep the normal code path at minimal indentation, by flipping this around so that non-success events immediately return nil.

Context: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
Done